### ISSUE TITLE
Guidance for DependencyContext followed by ICompilationAssemblyResolver usage

### ISSUE THREAD

**Issue by mwpowellhtx** [dotnet/runtime#3712]
Ostensibly, as I understand it, my process starts with adding a loaded ``Assembly`` to a ``DependencyContext`` and *Merging* that *context* with the original one. Upon adding the reference, I turn around and fold in a new ``CompositeCompilationAssemblyResolver`` with that ``AppBaseCompilationAssemblyResolver``. The inner resolver contains within it a ``PackageCompilationAssemblyResolver`` and a ``ReferenceAssemblyPathResolver``.

Ultimately this is to inform my *dotnet CLI* tooling, a [Code.Generation.Roslyn](/mwpowellhtx/Code.Generation.Roslyn) in which code generators, etc, may have internal project references, as well as external package references, and I am trying to do a bit of refactoring on my *Context/Resolver* assets in order to better isolate some test cases around that issue.

I am here just looking for a bit of guidance as to approaching that issue. Assuming the references have all been resolved, whether from packages or projects, etc, are there any others, I doubt it... Assuming all of that is squared away, I do not see any reason why my CG assemblies cannot be loaded and their generators run to generate code.

I look forward to your insights. Happy to elaborate as much as I can, but I think this is not so non-straightforward. Thanks!

**Comment by eerhardt** [comment]
Closing as the questions above haven't been answered. Feel free to re-open if you can provide more information.

**Comment by mwpowellhtx** [comment]
@eerhardt So you are saying that complex resolution contexts are handled seamlessly now? Because from where I was testing, this did not seem to be the case.

**Comment by eerhardt** [comment]
No, I'm not saying that complex resolution contexts are handled seamlessly now. I asked for more information on the scenario, but since it wasn't provided I was forced to close the issue. Feel free to re-open if you can provide more information. Sample code illustrating the issue or scenario would be super helpful.

**Comment by mwpowellhtx** [comment]
The scenario is as I described. I've got a dotnet CLI tooling that I want to invoke during a build. I have the plumbing all working, but it fails because the dependencies are not resolving correctly. Think, I've got something like a grammar parsed by an ANTLR based assembly, based on a Protobuf v2 specification, so on and so forth. This is/was using dotnet core 2.1 runtimes, I think, but admittedly it has been a few months since I was actively working in that area. I may dust it off and pull it forward to more recent runtime versions, but it is not on my front burner today.
